astropy/io/fits/hdu/compressed/compressed.py

```diff
@@ -364,6 +366,28 @@ def __init__(
         self._orig_bscale = self._bscale
         self._orig_bitpix = self._bitpix
 
+        if (
+            self._bitpix > 0
+            and "BLANK" not in self._header
+            and "ZBLANK" not in self._header
+        ):
+            # check for column named "ZBLANK"
+            for i in range(1, self._header["TFIELDS"] + 1):
+                if self._header[f"TTYPE{i}"] == "ZBLANK":
+                    # required BLANK keyword is missing
+                    # use most negative value as default
+                    self._header["BLANK"] = -(1 << (self._bitpix - 1))
+                    warnings.warn(
+                        "Setting default value for missing BLANK keyword in compressed extension",
```

Comment:
Shouldn't this warning also tell *what* default value was set so it can more easily be fixed upstream by users who also control the fits-writing code ?